### ISSUE TITLE
External commands with the backslash not recognized by Intellisense

### ISSUE THREAD

**Issue by hab12335** [James-Yu/LaTeX-Workshop#3695]
### Pre-check

- [X] The issue remains after disabling all other extensions and restarting Visual Studio Code.
- [X] The [FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) cannot address the issue.
- [X] If the issue is related to compiling a document, it can be successfully compiled in OS terminal.

### Extension Version

9.6.0

### Operating System and Version

Linux arch-laptop 6.1.9-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 01 Feb 2023 17:07:39 +0000 x86_64 GNU/Linux

### Visual Studio Code Version

1.75.0

### LaTeX Distribution Version

2022.63035-1

### Extension Environment

- [ ] I am using VSCodium.
- [ ] I am using the Snap or Flatpack version of Visual Studio Code.
- [ ] I am using LaTeX Workshop with Visual Studio Code Remote Containers.
- [ ] I am using LaTeX Workshop with Visual Studio Code Remote SSH.
- [ ] I am using LaTeX Workshop with Visual Studio Code Remote WSL.

### The Issue

Commands \inputed from external file are intellisensed without the backslaCh in main tex file.

### Reproduction Steps

Create .sty file, create new command, \input{myfile.sty}, then start with a backslaCh and begin writing the external command. Intellisense won't sense it. But if you omit the backslash, it shows up.

### Expected Behavior

backslaCh<my_external_command> should be Intellisensed 

### Actual Behavior

You need to omit the backslash and then go back to add it to have the desired behavior 

### LaTeX Workshop Output

[18:49:57][Logger] New log placeholder %WS1% registered for /home/hab12335/Documents/CV .
[18:49:57][Config] Configuration for workspace: file://%WS1% .
[18:49:57][Config] editor.acceptSuggestionOnEnter: "on" .
[18:49:57][Config] latex-workshop.bind.enter.key: true .
[18:49:57][Config] latex-workshop.docker.enabled: false .
[18:49:57][Config] latex-workshop.docker.image.latex: "" .
[18:49:57][Config] latex-workshop.hover.preview.mathjax.extensions: [] .
[18:49:57][Config] latex-workshop.intellisense.package.enabled: true .
[18:49:57][Config] latex-workshop.intellisense.update.aggressive.enabled: false .
[18:49:57][Config] latex-workshop.intellisense.update.delay: 1000 .
[18:49:57][Config] latex-workshop.latex.autoBuild.run: "onFileChange" .
[18:49:57][Config] latex-workshop.latex.build.forceRecipeUsage: true .
[18:49:57][Config] latex-workshop.latex.outDir: "%DIR%" .
[18:49:57][Config] latex-workshop.latex.recipes: [
 {
  "name": "lualatex",
  "tools": [
   "lualatex"
  ]
 }
] .
[18:49:57][Config] latex-workshop.latex.tools: [
 {
  "name": "lualatex",
  "command": "lualatex",
  "args": [
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "%DOC%"
  ]
 }
] .
[18:49:57][Config] latex-workshop.viewer.pdf.internal.keyboardEvent: "auto" .
[18:49:57][Manager] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
[18:49:57][Server] Creating LaTeX Workshop http and websocket server.
[18:49:58][Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":36855}
[18:49:58][Server] valdOrigin is http://127.0.0.1:36855
[18:49:58][Extension] Initializing LaTeX Workshop.
[18:49:58][Extension] Extension root: /home/hab12335/.vscode-oss/extensions/james-yu.latex-workshop-9.6.0-universal
[18:49:58][Extension] $PATH: /usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/opt/android-sdk/emulator:/home/hab12335/.dotnet/tools:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
[18:49:58][Extension] $SHELL: /bin/zsh
[18:49:58][Extension] $LANG: en_US.UTF-8
[18:49:58][Extension] $LC_ALL: undefined
[18:49:58][Extension] process.platform: linux
[18:49:58][Extension] process.arch: x64
[18:49:58][Extension] vscode.env.appName: Code - OSS
[18:49:58][Extension] vscode.env.remoteName: undefined
[18:49:58][Extension] vscode.env.uiKind: 1
[18:49:58][Extension] LaTeX Workshop initialized.
[18:49:58][Format][Bib] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[18:49:58][Extension] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
[18:49:58][Manager] Current workspace folders: ["file://%WS1%"]
[18:49:58][Manager] Found root file from active editor: %WS1%/main.tex
[18:49:59][Manager] Root file changed: from undefined to %WS1%/main.tex
[18:49:59][Manager] Start to find all dependencies.
[18:49:59][Manager] Root file languageId: latex
[18:49:59][Cacher][Watcher] Reset.
[18:49:59][Cacher] Adding %WS1%/main.tex .
[18:49:59][Cacher] Caching %WS1%/main.tex .
[18:49:59][Cacher] Input %WS1%/preamble.sty from %WS1%/main.tex .
[18:49:59][Cacher] Ignored %WS1%/preamble.sty .
[18:49:59][Cacher] Ignored %WS1%/preamble.sty .
[18:49:59][Cacher] Updated inputs of %WS1%/main.tex .
[18:49:59][Cacher][Watcher] Watched %WS1%/main.tex .
[18:49:59][DupLabel] Checking for duplicate labels: %WS1%/main.tex .
[18:49:59][Cacher] Updated elements of %WS1%/main.tex .
[18:49:59][Cacher] Updated bibs of %WS1%/main.tex .
[18:49:59][Cacher] Cached %WS1%/main.tex .
[18:49:59][Cacher][Path] Non-existent .fls for %WS1%/main.tex .
[18:50:02][Structure] Error loading cache during structuring: %WS1%/preamble.sty .
[18:50:53][Commander] BUILD command invoked.
[18:50:53][Commander] The document of the active editor: file://%WS1%/main.tex
[18:50:53][Commander] The languageId of the document: latex
[18:50:53][Manager] Current workspace folders: ["file://%WS1%"]
[18:50:53][Manager] Found root file from active editor: %WS1%/main.tex
[18:50:53][Manager] Keep using the same root file: %WS1%/main.tex
[18:50:53][Commander] Building root file: %WS1%/main.tex
[18:50:53][Builder] Build root file %WS1%/main.tex
[18:50:53][Builder] outDir: %WS1% .
[18:50:53][Builder] Preparing to run recipe: lualatex.
[18:50:53][Builder] Recipe step 1 The command is lualatex:["-synctex=1","-interaction=nonstopmode","-file-line-error","%WS1%/main"].
[18:50:53][Builder] env: undefined
[18:50:53][Builder] root: %WS1%/main.tex
[18:50:53][Builder] cwd: %WS1%
[18:50:53][Builder] LaTeX build process spawned with PID 10442.
[18:50:57][Parser][TexLog] Logged 2 messages.
[18:50:57][Builder] Finished a step in recipe with PID 10442.
[18:50:57][Builder] Successfully built %WS1%/main.tex .
[18:50:57][Viewer] Call refreshExistingViewer: {"sourceFile":"%WS1%/main.tex"}
[18:50:57][Viewer] Not found PDF viewers to refresh: undefined
[18:50:57][Cacher][Path] Non-existent .fls for %WS1%/main.tex .

**Comment by jlelong** [comment]
We do not parse `.sty` file as their syntax is usually far too complicated. If you want to define commands in an external file, use a `.tex` file.
